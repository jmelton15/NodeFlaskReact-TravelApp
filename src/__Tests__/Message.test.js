import { fireEvent, render } from '@testing-library/react';
import {user,message1} from "./TestHelperObjects";
import Message from "../Components/Messages/message";


// simple smoke test for Message component
it("renders a Message component without crashing", function() {
    render(<Message currentUser={1} message={message1}/>);
});

// simple snapshot test for Message Component
it("should match snapshot for Message", function() {
    const {asFragment} = render (<Message currentUser={1} message={message1}/>);
    expect(asFragment()).toMatchSnapshot();
});

it("should display correct ownership of message (this case should be own)",() => {
    const {getByTestId} = render(<Message currentUser={1} message={message1}/>)
    const ownership = getByTestId("messageOwnership");
    expect(ownership).toHaveClass('own');
})