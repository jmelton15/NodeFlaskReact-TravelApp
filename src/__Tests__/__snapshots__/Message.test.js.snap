// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot for Message 1`] = `
<DocumentFragment>
  <div
    class="Message other"
    data-testid="messageOwnership"
  >
    <img
      alt="Avatar Pic"
      class="Message-AvatarPic"
      src="some pic route"
    />
    <div
      id="message-body"
    >
      <p
        class="Message-Text-Other"
        id="message-text"
      >
        Hello From Test1
      </p>
    </div>
  </div>
</DocumentFragment>
`;
